cli/tests/pcluster/validators/test_all_validators.py

```diff
@@ -295,3 +300,296 @@ def test_validators_are_called_with_correct_argument(test_datadir, mocker):
     ebs_volume_size_snapshot_validator.assert_called()
     shared_ebs_volume_id_validator.assert_called()
     fsx_persistent_options_validator.assert_called()
+
+
+def test_scheduler_plugin_all_validators_are_called(test_datadir, mocker):
+    """Verify that all validators are called during validation."""
+    mockers = []
+    _mock_all_validators(mocker, mockers, additional_modules=[scheduler_plugin_validators])
+
+    # mock properties that use boto3 calls
+    mocker.patch(
+        "pcluster.config.cluster_config.HeadNode.architecture", new_callable=PropertyMock(return_value="x86_64")
+    )
+    mocker.patch(
+        "pcluster.config.cluster_config.SchedulerPluginComputeResource.architecture",
+        new_callable=PropertyMock(return_value="x86_64"),
+    )
+    mocker.patch(
+        "pcluster.config.cluster_config.HeadNodeNetworking.availability_zone",
+        new_callable=PropertyMock(return_value="us-east-1a"),
+    )
+    mocker.patch(
+        "pcluster.config.cluster_config.BaseClusterConfig.head_node_ami",
+        new_callable=PropertyMock(return_value="ami-12345678"),
+    )
+    mocker.patch(
+        "pcluster.config.cluster_config.SchedulerPluginClusterConfig.get_instance_types_data",
```

Comment:
Can we put these in to a function and reuse it in the slurm test?

Reply:
Here some mocker patch are different from slurm and plugin 
`SchedulerPluginComputeResource. architecture ` and `SchedulerPluginClusterConfig. get_instance_types_data`, they can not be reused in slurm test.
Only the fields from the base config can be reused